File: app_vlxd/lib/model/KhachHang.dart
class KhachHang{
  String? maKH;
  String tenKH;
  String userName;
  String email;
  String diaChi;
  String gioiTinh;
  String ngaySinh;
  String dienThoai;
  bool? active;
  String? password;

  KhachHang({
    this.maKH,
    required this.tenKH,
    required this.userName,
    required this.email,
    required this.diaChi,
    required this.gioiTinh,
    required this.ngaySinh,
    required this.dienThoai,
    this.active,
    this.password,
  });

  factory KhachHang.fromJson(Map<String, dynamic> json) {
    return KhachHang(
      maKH: json['maKH'],
      tenKH: json['tenKH'],
      userName: json['userName'],
      email: json['email'],
      diaChi: json['diaChi'],
      gioiTinh: json['gioiTinh'],
      ngaySinh: json['ngaySinh'],
      dienThoai: json['dienThoai'],
      active: json['active'],
      password: "",
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'tenKH': tenKH,
      'userName': userName,
      'email': email,
      'pass': password,
      'diaChi': diaChi,
      'gioiTinh': gioiTinh,
      'ngaySinh': ngaySinh,
      'dienThoai': dienThoai,
    };
  }

}

File: vatlieuxaydung/src/main/java/com/example/vatlieuxaydung/model/KhachHang.java
package com.example.vatlieuxaydung.model;

import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.FieldDefaults;

import java.time.LocalDate;


@Entity
@Getter
@Setter
@NoArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
@Table(name = "khachhang")
public class KhachHang {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(name = "ma_KH", columnDefinition = "CHAR(36)")
    String maKH;
    @Column(name = "tenKh")
    String tenKH;
    @Column(name = "userName")
    String userName;
    @Column(name = "email")
    String email;
    @Column(name = "pass")
    String pass;
    @Column(name = "diaChi")
    String diaChi;
    @Column(name = "gioiTinh")
    String gioiTinh;
    @Column(name = "NSinh")
    LocalDate ngaySinh;
    @Column(name = "dienThoai")
    String dienThoai;
    @Column(name = "active")
    boolean active;

    public KhachHang(String maKH, String tenKH, String userName, String email, String pass, String diaChi,
                     String gioiTinh, LocalDate ngaySinh, String dienThoai, boolean active) {
        this.maKH = maKH;
        this.tenKH = tenKH;
        this.userName = userName;
        this.email = email;
        this.pass = pass;
        this.diaChi = diaChi;
        this.gioiTinh = gioiTinh;
        this.ngaySinh = ngaySinh;
        this.dienThoai = dienThoai;
        this.active = active;
    }

    public KhachHang(String tenKH, String userName, String email, String pass, String diaChi,
                     String gioiTinh, LocalDate ngaySinh, String dienThoai, boolean active) {
        this.tenKH = tenKH;
        this.userName = userName;
        this.email = email;
        this.pass = pass;
        this.diaChi = diaChi;
        this.gioiTinh = gioiTinh;
        this.ngaySinh = ngaySinh;
        this.dienThoai = dienThoai;
        this.active = active;
    }
}
